libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewEventDispatcher.java

```diff
@@ -167,14 +173,22 @@ void onInstructionListVisibilityChanged(boolean shown) {
 
   SpeechAnnouncement onAnnouncement(SpeechAnnouncement announcement) {
     if (speechAnnouncementListener != null) {
-      return speechAnnouncementListener.willVoice(announcement);
+      String textAnnouncement = speechAnnouncementListener.willVoice(announcement);
+      SpeechAnnouncement announcementToBeVoiced = SpeechAnnouncement.builder().announcement(textAnnouncement).build();
+      return announcementToBeVoiced;
     }
     return announcement;
   }
 
   BannerInstructions onBannerDisplay(BannerInstructions instructions) {
     if (bannerInstructionsListener != null) {
-      return bannerInstructionsListener.willDisplay(instructions);
+      String textBanner = bannerInstructionsListener.willDisplay(instructions);
+      List<BannerComponents> bannerComponents = new ArrayList<>(ONE);
+      bannerComponents.add(instructions.primary().components().get(FIRST).toBuilder().text(textBanner)
+        .abbreviation(textBanner).build());
```

Comment:
you don't have to set an abbreviation, and there shouldn't be one unless there's an `abbreviationPriority` as well.

Reply:
Nah, if not included the banner is not overridden at all. This is because how the final banner text to display is generated. We should revisit that logic.